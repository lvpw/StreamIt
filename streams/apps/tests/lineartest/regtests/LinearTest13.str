/** 
 * Test recognizing the special "Identity" filters 
 * that are recognized as a special stream construct.
 **/
void->void pipeline LinearTest13 {
  add FloatSource();
  add LinearPipe();
  add FloatSink();
}

void->float stateful filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** Pipeline of linear filters which have constant components. **/
float->float pipeline LinearPipe {
  add Identity<float>();
  add FloatFilter1();
}

/** filter that computes [[1 2][3 4][5 6][7 8]] + [9 10] **/
float->float filter FloatFilter1 {
  work pop 4 push 2 {
    float t1 = pop();
    float t2 = pop();
    float t3 = pop();
    float t4 = pop();
    push(2*t4 + 4*t3 + 6*t2 + 8*t1 + 10);
    push(1*t4 + 3*t3 + 5*t2 + 7*t1 + 9);
  }
}
