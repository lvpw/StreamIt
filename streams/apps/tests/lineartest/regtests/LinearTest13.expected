FloatFilter1_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]
 [5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]]
FloatFilter1_1.vector
[[9.0+0.0i 10.0+0.0i]]
Identity_1.matrix
[[1.0+0.0i]]
Identity_1.vector
[[0.0+0.0i]]
LinearPipe_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]
 [5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]]
LinearPipe_1.vector
[[9.0+0.0i 10.0+0.0i]]
